### PR TITLE
Allow table to receive an empty state instance and show it if no items are present

### LINKED ISSUE #826: Add non-ideal/empty state display to table
We should add support to `Table` to customize what should be displayed when there are no rows in the data set. Possibly this could just be an `EmptyState` component?

### PR DESCRIPTION
**Pull request by francocorreasosa**
## 🆕 Added  
- `emptyMessage` prop: Message to display in case there are no rows in the table. (Defaults to "There are no items to display"

> Resolves #826

### CODE REVIEW

**Comment by landitus** [comment]
Ops, I see this in chromatic:

screen shot 2018-08-23 at 16 02 38

**Comment by landitus** [comment]
This is how it should look :) Sorry I didn't clarify !!

The idea is to keep the Headers
But show a configurable message 

screen shot 2018-08-23 at 16 06 44

**Comment by francocorreasosa** [comment]
No problem at all!

**Comment by francocorreasosa** [comment]
Done! @landitus gan you take a look again?

File: core/components/molecules/table/table.js [hunk 1]
```diff
@@ -173,6 +176,20 @@ Table.Cell = styled.td`
   width: ${props => props.column.width || 'auto'};
 `
 
+Table.EmptyState = ({ rows, children }) => {
+  if (rows.length > 0 || !children) return null
+
+  const TableEmptyState = styled.div`
+    padding: ${spacing.small};
+    background-color: rgb(250, 250, 250);
+    border-radius: 5px;
```
Comment (landitus) [inline core/components/molecules/table/table.js]: Let's change this to use the common border radius token `$misc.radius`

File: core/components/molecules/table/table.js [hunk 2]
```diff
@@ -173,6 +176,20 @@ Table.Cell = styled.td`
...
+    text-align: center;
+    margin-top: ${spacing.xsmall};
+  `
```
Comment (landitus) [inline core/components/molecules/table/table.js]: Let's add a `color: $colors.text.default` rule so the typography uses the default text color

File: core/components/molecules/table/table.js (hunk 1 above)
Comment (francocorreasosa) [reply core/components/molecules/table/table.js]: Ohh nice catch, thanks!

Comment (francocorreasosa) [reply core/components/molecules/table/table.js]: Done

**Comment by landitus** [comment]
I was trying the example [shown in the docs ](https://auth0-cosmos-table-empty-state.now.sh/docs/#/component/table#empty-table)in the Playground and got errors till I left it like[ this example](https://auth0-cosmos-table-empty-state.now.sh/docs/#/playground?code=%250A/*%20you%20have%20access%20to%20all%20cosmos%20components%20here%20*/%250A%250Aconst%20Example%20=%20()%20=%3E%20%7B%250A%20%20return%20(%250A%20%20%20%20%3CTable%20items=%7B%5B%5D%7D%20%3E%20%20%250A%20%20%20%20%3CTable.Column%20field=%22image%22%20width=%2250px%22%3E%3C/Table.Column%3E%250A%20%20%20%20%3CTable.Column%20field=%22name%22%20title=%22Name%22%20width=%2230%25%22%20/%3E%250A%20%20%20%20%3CTable.Column%20field=%22country%22%20title=%22Country%22%20/%3E%250A%20%20%20%20%3CTable.Column%20field=%22goals%22%20title=%22Goals%22%20/%3E%250A%20%20%20%20%3CTable.Column%20field=%22assists%22%20title=%22Assists%22%20/%3E%250A%20%20%20%20%3C/Table%3E%20%250A%20%20)%250A%7D%250A%250Arender(Example)%250A)

I get errors because we are trying to render the item in the Avatar column and also the sortable props gives error. @francocorreasosa Can you check if the docs example needs review? Or how would we deal with a sortable column that is empty?

**Comment by landitus** [comment]
Disregard my last comment, looks to be due to the Playground and not a problem with the component

**Pull request merged**